package/network/services/hostapd/src/src/ap/ubus.c

```diff
@@ -1269,3 +1269,30 @@ void hostapd_ubus_notify(struct hostapd_data *hapd, const char *type, const u8 *
 
 	ubus_notify(ctx, &hapd->ubus.obj, type, b.head, -1);
 }
+
+void hostapd_ubus_notify_beacon_report(struct hostapd_data *hapd, const u8 *addr, u8 token, u8 rep_mode, const u8 *pos, size_t len)
+{
+	if (!hapd->ubus.obj.has_subscribers)
+		return;
+
+	if (!addr)
+		return;
+
+	if (len < 25)
```

Comment:
Try to avoid magic numbers.

```c
if (len < sizeof(struct rrm_measurement_beacon_report))
    return;
```

this check should be handled in `hostapd_handle_beacon_report` and `hostapd_ubus_notify_beacon_report` should get pointer to the `struct rrm_measurement_beacon_report`.